proxy/http/remap/RemapPluginInfo.cc

```diff
@@ -271,6 +273,31 @@ RemapPluginInfo::indicatePostReload(TSRemapReloadStatus reloadStatus)
   resetPluginContext();
 }
 
+/* Initialize strategy (optional). */
+std::shared_ptr<TSNextHopSelectionStrategy>
+RemapPluginInfo::initStrategy(void *ih)
+{
+  if (!init_strategy_cb) {
+    PluginDebug(_tag, "plugin '%s' has no init_strategy_cb, returning nullptr", _configPath.c_str());
+    return std::shared_ptr<TSNextHopSelectionStrategy>();
+  }
+  char tmpbuf[2048];
+  ink_zero(tmpbuf);
+  TSNextHopSelectionStrategy *strategy_raw = nullptr;
+  if (init_strategy_cb(strategy_raw, ih, tmpbuf, sizeof(tmpbuf) - 1) != TS_SUCCESS) {
```

Comment:
How does `strategy_raw` become not `nullptr`?

Reply:
`init_strategy_cb` is `TSRemapInitStrategy`, which takes a `TSNextHopSelectionStrategy *&`. A plugin sets that pointer-ref. https://github.com/apache/trafficserver/blob/00e236d94aa5577e8a1d0f94d18787bd2c7e85b3/plugins/experimental/nexthop_strategy_consistenthash/plugin_consistenthash.cc#L74